frontend/packages/console-shared/src/selectors/node.ts

```diff
@@ -1,68 +1,56 @@
-import * as _ from 'lodash';
-import { NodeKind } from '@console/internal/module/k8s';
-import { NodeAddress, NodeCondition } from '../types';
+import {
+  NodeAddress,
+  NodeKind,
+  Taint,
+} from '@console/dynamic-plugin-sdk/src/extensions/console-types';
 
 const NODE_ROLE_PREFIX = 'node-role.kubernetes.io/';
 
-export const getNodeRoles = (node: NodeKind): string[] => {
-  const labels = _.get(node, 'metadata.labels');
-  return _.reduce(
-    labels,
-    (acc: string[], v: string, k: string) => {
-      if (k.startsWith(NODE_ROLE_PREFIX)) {
-        acc.push(k.slice(NODE_ROLE_PREFIX.length));
-      }
-      return acc;
-    },
-    [],
-  );
-};
+export const getNodeRoles = (node: NodeKind): string[] =>
+  Object.keys(node?.metadata?.labels ?? {}).reduce((acc: string[], k) => {
+    if (k.startsWith(NODE_ROLE_PREFIX)) {
+      acc.push(k.slice(NODE_ROLE_PREFIX.length));
+    }
+    return acc;
+  }, []);
 
 export const getNodeRole = (node: NodeKind): string =>
   getNodeRoles(node).includes('control-plane') ? 'control-plane' : 'worker';
 
-export const getNodeRoleMatch = (node: NodeKind, role: string): boolean => {
-  const roles = getNodeRoles(node);
-
-  return roles.filter((elem) => elem === role).length > 0;
-};
+export const getNodeRoleMatch = (node: NodeKind, role: string): boolean =>
+  getNodeRoles(node).filter((elem) => elem === role).length > 0;
 
-export const getNodeAddresses = (node: NodeKind): NodeAddress[] =>
-  _.get(node, 'status.addresses', []);
+export const getNodeAddresses = (node: NodeKind): NodeAddress[] => node?.status?.addresses ?? [];
 
-type NodeMachineAndNamespace = {
-  name: string;
-  namespace: string;
-};
-export const getNodeMachineNameAndNamespace = (node: NodeKind): NodeMachineAndNamespace => {
-  const machine = _.get(node, 'metadata.annotations["machine.openshift.io/machine"]', '/');
+export const getNodeMachineNameAndNamespace = (node: NodeKind): [string, string] => {
```

Comment:
Nit: we can use TypeScript named tuple elements feature here

```suggestion
export const getNodeMachineNameAndNamespace = (node: NodeKind): [name: string, namespace: string] => {
```